### PR TITLE
docs(operators): add documentation for operator mergeMap

### LINKED ISSUE #119: Add docs for operator : transformation - mergeMap
Add docs for operator mergeMap under group transformation 
 
Below is minimum details needed 
 
- [ ] name 
- [ ] operatorType 
- [ ] signature 
- [ ] parameters 
- [ ] short description 
- [ ] Technical description 
- [ ] Visual example  (Marble diagram) 
- [ ] Real world example (Non-framework) 
- [ ] Related Operators

### PR DESCRIPTION
**Pull request by natmegs**
Add initial version of documentation for transformation operator 'mergeMap'
closes #119

### CODE REVIEW

**Comment by natmegs** [comment]
Whoops, looking at the issue I didn't see that there was already a PR in the works. Disregard this one!

**Comment by ladyleet** [comment]
Thx @natmegs!

File: src/operator-docs/transformation/mergeMap.ts [hunk 1]
```diff
@@ -1,6 +1,79 @@
...
+      code: `
+        const letters = Rx.Observable.of('a', 'b', 'c');
+        const result = letters.mergeMap(x =>
+          Rx.Observable.interval(1000).map(i => x+i)
+        );
+        result.subscribe(x => console.log(x));
+      `,
+      externalLink: {
+        platform: 'JSBin',
+        url: 'http://jsbin.com/pekelowibe/edit?js,console'
```
Comment (ashwin-sureshkumar) [inline src/operator-docs/transformation/mergeMap.ts]: please use `embed` instead of `edit`

File: src/operator-docs/transformation/mergeMap.ts [hunk 2]
```diff
@@ -1,6 +1,85 @@
...
+        import { interval } from 'rxjs/observable/interval';
+        import { mergeMap, map } from 'rxjs/operators';
+
+        const letters = of('a', 'b', 'c');
+        const result = letters.pipe(
+                            mergeMap(x => {
+                              return interval(1000).pipe(map(i => x+i));
+                            })
+                       );
+        result.subscribe(x => console.log(x));
```
Comment (sumitarora) [inline src/operator-docs/transformation/mergeMap.ts]: Could you please add expected output of jsbin as comments.

**Pull request merged**